src/components/idle-picker/IdlePicker.js

```diff
@@ -0,0 +1,193 @@
+import React, { Component } from 'react';
+import { default as RcSlider } from 'rc-slider';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Display from './display/Display';
+
+import styles from './IdlePicker.css';
+import 'rc-slider/assets/index.css';
+
+const marks = {
+    1: '01',
+    2: '',
+    3: '',
+    4: '',
+    5: '',
+    6: '',
+    7: '',
+    8: '',
+    9: '',
+    10: '10',
+};
+
+export default class IdlePicker extends Component {
+    static propTypes = {
+        defaultValue: PropTypes.number,
+        handlePickerChange: PropTypes.func,
+    };
+
+    static defaultProps = {
+        defaultValue: 3,
+    };
+
+    constructor(props) {
+        super(props);
+
+        let { defaultValue } = this.props;
+
+        if (defaultValue > 10 || defaultValue < 1) {
+            defaultValue = 3;
+        }
+
+        this.state = {
+            displayValue: defaultValue,
+            pickerValue: defaultValue,
+            forcePickerValue: true,
+        };
+
+        this.componentRef = React.createRef();
+        this.previousHandleWasBeforeChange = false;
+    }
+
+    componentDidMount() {
+        const picker = this.componentRef.current;
+        const handle = picker.querySelector('.rc-slider-handle');
+        const track = picker.querySelector('.rc-slider-track');
+
+        this.handleClick = false;
+        this.mouseUpOnElement = false;
+
+        const appendClass = () => {
```

Comment:
This is fine, but I would consider defining these as class methods.

Reply:
Though @paulobmarcos did not write it here, he also gave me that feedback. I've made these changes for all those methods in componentDidMount